beacon-chain/sync/initial-sync/blocks_fetcher.go

```diff
@@ -405,7 +408,10 @@ func (f *blocksFetcher) waitForMinimumPeers(ctx context.Context) ([]peer.ID, err
 			return nil, ctx.Err()
 		}
 		headEpoch := helpers.SlotToEpoch(f.headFetcher.HeadSlot())
-		_, _, peers := f.p2p.Peers().BestFinalized(params.BeaconConfig().MaxPeersToSync, headEpoch)
+		_, _, peers, err := f.p2p.Peers().BestFinalized(params.BeaconConfig().MaxPeersToSync, headEpoch)
+		if err != nil {
+			log.WithError(err).Debug("Could not determine best finalized")
```

Comment:
we will panic below with a nil slice, we need to again wait for enough suitable peers.

Reply:
Again, imo, this shows that moving `len(peers)` check to the `BestFinalized` wasn't enough motivated. Below I check the `len(peers)` and continue waiting if they are not enough.

Overall, as I've mentioned earlier, `BestFinalized()` may need rewriting: but as it is, client code is expected to check the length of peers before using any peer's related extra info returned by that method. I don't totally buy the idea expressed in QSP-45 - that it leads to undetermined behaviour, as using root w/o checking is a misuse.